client/lib/cart-values/cart-items.ts

```diff
@@ -130,6 +130,17 @@ export function has100YearPlan( cart: ObjectWithProducts ): boolean {
 	return getAllCartItems( cart ).some( is100Year );
 }
 
+const isHundredYearDomain = ( product: {
+	extra?: RequestCartProductExtra;
+	volume?: number;
+} ): boolean => {
+	return Boolean( product.extra?.is_hundred_year_domain || product.volume === 100 );
+};
```

Comment:
This logic was inspired by https://github.com/Automattic/wp-calypso/blob/30624f57ddcd8779dbd2ad55ce45064d8ddd9522/client/landing/stepper/declarative-flow/hundred-year-domain.ts#L60